cmd/eksctl-anywhere/cmd/options.go

```diff
@@ -166,11 +168,57 @@ func buildCliConfig(clusterSpec *cluster.Spec) *config.CliConfig {
 	return cliConfig
 }
 
-func buildCreateCliConfig(clusterOptions *createClusterOptions) config.CreateClusterCLIConfig {
-	createCliConfig := config.CreateClusterCLIConfig{}
+func buildCreateCliConfig(clusterOptions *createClusterOptions) (*config.CreateClusterCLIConfig, error) {
+	createCliConfig := &config.CreateClusterCLIConfig{}
 	createCliConfig.SkipCPIPCheck = clusterOptions.skipIpCheck
+	if clusterOptions.noTimeouts {
+		maxTime := time.Duration(math.MaxInt64)
+		createCliConfig.NodeStartupTimeout = maxTime
+		createCliConfig.UnhealthyMachineTimeout = maxTime
 
-	return createCliConfig
+		return createCliConfig, nil
+	}
+
+	unhealthyMachineTimeout, err := time.ParseDuration(clusterOptions.unhealthyMachineTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeStartupTimeout, err := time.ParseDuration(clusterOptions.nodeStartupTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	createCliConfig.NodeStartupTimeout = nodeStartupTimeout
+	createCliConfig.UnhealthyMachineTimeout = unhealthyMachineTimeout
+
+	return createCliConfig, nil
+}
+
+func buildUpgradeCliConfig(clusterOptions *upgradeClusterOptions) (*config.UpgradeClusterCLIConfig, error) {
+	upgradeCliConfig := config.UpgradeClusterCLIConfig{}
+	if clusterOptions.noTimeouts {
+		maxTime := time.Duration(math.MaxInt64)
+		upgradeCliConfig.NodeStartupTimeout = maxTime
+		upgradeCliConfig.UnhealthyMachineTimeout = maxTime
+
+		return &upgradeCliConfig, nil
+	}
+
+	unhealthyMachineTimeout, err := time.ParseDuration(clusterOptions.unhealthyMachineTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	nodeStartupTimeout, err := time.ParseDuration(clusterOptions.nodeStartupTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeCliConfig.NodeStartupTimeout = nodeStartupTimeout
+	upgradeCliConfig.UnhealthyMachineTimeout = unhealthyMachineTimeout
+
+	return &upgradeCliConfig, nil
 }
```

Comment:
this code is mostly copy-paste from the other method. not a blocker, but we should consider reusing the code

Reply:
yes i have a follow up to combine the timeout structs for create and upgrade defaulters.